opsgenie/resource_opsgenie_service_incident_rule.go

```diff
@@ -0,0 +1,411 @@
+													Type:         schema.TypeString,
+													Optional:     true,
+													ValidateFunc: validation.StringLenBetween(1, 15000),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceOpsGenieServiceIncidentRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := service.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+
+	service_id := d.Get("service_id").(string)
+	createRequest := &service.CreateIncidentRuleRequest{
+		ServiceId: service_id,
+	}
+
+	//if len(d.Get("conditions").([]interface{})) > 0 {
+	incident_rule := d.Get("incident_rule").([]interface{})
+	for _, v := range incident_rule {
+		config := v.(map[string]interface{})
+		createRequest.ConditionMatchType = og.ConditionMatchType(config["condition_match_type"].(string))
+		createRequest.Conditions = expandOpsGenieServiceIncidentRuleConditions(config["conditions"].([]interface{}))
+		createRequest.IncidentProperties = expandOpsGenieServiceIncidentRuleIncidentProperties(config["incident_properties"].([]interface{}))
+	}
+	//}
+
+	log.Printf("[INFO] Creating OpsGenie Service Incident Rule for service '%s'", d.Get("service_id").(string))
+	result, err := client.CreateIncidentRule(context.Background(), createRequest)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(result.Id)
+
+	return resourceOpsGenieServiceIncidentRuleRead(d, meta)
+}
+
+func resourceOpsGenieServiceIncidentRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := service.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	service_id := d.Get("service_id").(string)
+	incident_rule_id := d.Id()
+
+	log.Printf("[INFO] Reading OpsGenie Service Incident Rule for service: '%s' for rule ID: '%s'", service_id, incident_rule_id)
+
+	incident_rule_res, err := client.GetIncidentRules(context.Background(), &service.GetIncidentRulesRequest{
+		ServiceId: service_id,
+	})
+	if err != nil {
+		x := err.(*ogClient.ApiError)
+		if x.StatusCode == 404 {
+			log.Printf("[WARN] Removing Service Incident Rule because it's gone %s", service_id)
+			d.SetId("")
+			return nil
+		}
+	}
+
+	for _, v := range incident_rule_res.IncidentRule {
+		if v.Id == incident_rule_id {
+			d.Set("service_id", service_id)
+			d.Set("incident_rule", flattenOpsGenieServiceIncidentRules(v))
+		}
+	}
+	return nil
+}
+
+func resourceOpsGenieServiceIncidentRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, err := service.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+
+	service_id := d.Get("service_id").(string)
+	incident_rule_id := d.Id()
+
+	updateRequest := &service.UpdateIncidentRuleRequest{
+		ServiceId:      service_id,
+		IncidentRuleId: incident_rule_id,
+	}
+
+	//if len(d.Get("conditions").([]interface{})) > 0 || len(d.Get("incident_properties").([]interface{})) > 0 {
```

Comment:
do we need this ?

Reply:
incident_properties is a mandatory field. So, No need to check this